internal/notif/ntfy/client.go

```diff
@@ -0,0 +1,100 @@
+package ntfy
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"path"
+	"strconv"
+
+	"github.com/crazy-max/diun/v4/internal/model"
+	"github.com/crazy-max/diun/v4/internal/msg"
+	"github.com/crazy-max/diun/v4/internal/notif/notifier"
+)
+
+// Client represents an active ntfy notification object
+type Client struct {
+	*notifier.Notifier
+	cfg  *model.NotifNtfy
+	meta model.Meta
+}
+
+// New creates a new ntfy notification instance
+func New(config *model.NotifNtfy, meta model.Meta) notifier.Notifier {
+	return notifier.Notifier{
+		Handler: &Client{
+			cfg:  config,
+			meta: meta,
+		},
+	}
+}
+
+// Name returns notifier's name
+func (c *Client) Name() string {
+	return "ntfy"
+}
+
+// Send creates and sends a ntfy notification with an entry
+func (c *Client) Send(entry model.NotifEntry) error {
+	hc := http.Client{
+		Timeout: *c.cfg.Timeout,
+	}
+
+	message, err := msg.New(msg.Options{
+		Meta:          c.meta,
+		Entry:         entry,
+		TemplateTitle: c.cfg.TemplateTitle,
+		TemplateBody:  c.cfg.TemplateBody,
+	})
+	if err != nil {
+		return err
+	}
+
+	title, body, err := message.RenderMarkdown()
+	if err != nil {
+		return err
+	}
+
+	u, err := url.Parse(c.cfg.Endpoint)
+	if err != nil {
+		return err
+	}
+	u.Path = path.Join(u.Path, c.cfg.Topic)
+
+	q := u.Query()
+	u.RawQuery = q.Encode()
+
+	req, err := http.NewRequest("POST", u.String(), bytes.NewBuffer(body))
+	if err != nil {
+		return err
+	}
+
+	req.Header.Set("Content-Type", "text/plain")
+	req.Header.Add("Content-Length", strconv.Itoa(len(string(body))))
+	req.Header.Set("User-Agent", c.meta.UserAgent)
+	req.Header.Set("Tags", c.cfg.Emoji)
+	req.Header.Set("Priority", c.cfg.Priority)
```

Comment:
I think it would be better to use the same parameter names provided by the API: https://docs.ntfy.sh/publish/#list-of-all-parameters

For example
* `Emoji` should be `Tags` and `tags` in yaml config.
* `Priority` looks good

I would also prefer to use the well-known parameter header like `X-Tags` or `X-Priority`.